csrc/host_ir/executor.cpp

```diff
@@ -469,6 +469,22 @@ void HostIrEvaluator::handle(MatmulOp* matmul) {
   }
 }
 
+void HostIrEvaluator::handle(kir::Allocate* allocate) {
+  NVF_ERROR(
+      allocate->buffer()->isA<TensorView>(),
+      "Allocation must be on a TensorView but got ",
+      allocate->buffer());
+  TensorView* tv = allocate->buffer()->as<TensorView>();
+  GlobalBufferInfo info =
+      getBufferInfos(expr_evaluator_, PrimDataType::Int, {tv}).at(0);
+  AliasInfo alias_info = {
+      .type = AllocationType::New, .aliased_io = nullptr, .hide_output = false};
+  c10::Device device =
+      communicator_ ? communicator_->device() : at::Device("cuda:0");
+  at::Tensor tensor = allocateTensor(info, alias_info, device, expr_evaluator_);
+  expr_evaluator_.bind(tv, tensor);
```

Comment:
Is an Allocate allowed in a for-loop? If yes, can you test that? I suspect something might break because Allocate doesn't consume the loop index and might not be properly invalidated. This might cause its direct consumers, if evaluated by ExprEvaluator, to fail evaluate_validate: https://github.com/NVIDIA/Fuser/blob/736a541348af6121aa91f8164209d6815db2d092/csrc/host_ir/executor.cpp#L486

Reply:
> Is an Allocate allowed in a for-loop? If yes, can you test that?

Allocating in a for-loop is supported (even though probably not recommended perf-wise). I added a test for that.

> I suspect something might break because Allocate doesn't consume the loop index and might not be properly invalidated.

Indeed, it won't get invalidated, as it shouldn't, since it doesn't use `i`.

> This might cause its direct consumers, if evaluated by ExprEvaluator, to fail evaluate_validate:
https://github.com/NVIDIA/Fuser/blob/736a541348af6121aa91f8164209d6815db2d092/csrc/host_ir/executor.cpp#L486

Agreed, but somehow the test passes... Maybe we got "lucky" and the `at::Tensor::is_same` (which is used in `evaluate_validate`) returned `true` this time but might fail in other cases, idk. Anyway, I think we should actually remove `evaluate_validate`, which I added for debugging purposes but it's not required and doesn't make a lot of sense in a non-ssa program. Lmk